### PR TITLE
Allow frame number codes f, m, l for first, middle or last

### LINKED ISSUE #4709: For movie -M, let have a letter to indicate last frame
I am doing an animation with movie with date frames. I am want to make the single master frame ([-M](https://docs.generic-mapping-tools.org/dev/movie.html#m) from the last one. Since I am using dates (like animation 8) it is not easy to know the its number. 

I was thinking to have something like:

 `movie -M+l`

  

 
I think it is not possible so I asked the feature directly.

**Are you willing to help implement and maintain this feature?**  I wish I know how.

### PR DESCRIPTION
**Pull request by PaulWessel**
This addresses the reqest in #4709 when it is not easy to know the number of the last (or middle) frame.  Closes #4709.  @Esteban82, could you please give this a try before we approve and merge?

### CODE REVIEW

**Comment by Esteban82** [comment]
> @Esteban82, could you please give this a try before we approve and merge?

Yes, if someone tell me how could I do it I would be glad to help.

**Comment by meghanrjones** [comment]
> > @Esteban82, could you please give this a try before we approve and merge?
> 
> Yes, if someone tell me how could I do it I would be glad to help.

Do you have git installed? I can give you some instructions for building the branch if you do.

**Comment by Esteban82** [comment]
Yes, please. I have git 2.25.1

**Comment by meghanrjones** [comment]
Assuming that you have previously [cloned the gmt repository](https://github.com/GenericMappingTools/gmt/blob/master/BUILDING.md#getting-gmt-source-codes) and [configured the cmake files](https://github.com/GenericMappingTools/gmt/blob/master/BUILDING.md#configuring):
1. `cd` into your cloned gmt repository
2. Pull down the latest changes by running `git pull`
3. Run `git branch -r` and confirm that `origin/movie-master-frame` is included in the list of remote branches
4. If yes, run `git checkout -b movie-master-frame origin/movie-master-frame`, which will create a new local branch and switch to it
5. If you already have the cmake files set up from previously building gmt from source, you can follow the typical [build and install steps](https://github.com/GenericMappingTools/gmt/blob/master/BUILDING.md#building-gmt-source-codes) (usually ```cd build; cmake --build .; cmake --build . --target install```

Afterwards, you can switch back to the main branch by running ```git checkout master``` and (optionally) delete the movie-master-frame branch using ```git branch -d movie-master-frame```

**Comment by Esteban82** [comment]
I' ve never managed to do it. But I will try tomorrow again.

**Comment by Esteban82** [comment]
> Run `git branch -r` and confirm that `origin/movie-master-frame` is included in the list of remote branches

I have no that brunch. When I run it I have the following options: 

```
$ git branch -r
  origin/HEAD -> origin/master
  origin/master
```

**Comment by joa-quim** [comment]
Federico, if you use Visual Studio Code is much simpler. I use a mix of it and the TortoiseGit.
With VSC if you load the entire ``src`` directory and look at the lower left corner you'll see the branch name. Click on it and just scroll the list. You can also create new branches here very easily.
image

**Comment by Esteban82** [comment]
Thanks @joa-quim. Yes, I heard that with VSC it is easier. It's on my to-do list.

**Comment by Esteban82** [comment]
Finally I manage to test it (I think). I couldn't get the last frame (-Ml,png). 
I get the following message:
`movie_master.sh: línea 6: movie_params_-1.sh: No existe el archivo o el directorio`
And I get a different frame without label (-Lc).

I am not sure if I installed the branch correctly. It the first time that I build from source and working with git.
The gmt version is: Version 6.2.0_b7eaa8f_2021.01.29 [64-bit]

**Comment by PaulWessel** [comment]
Thanks for trying it @Esteban82.  I will investigate.

**Comment by PaulWessel** [comment]
I had some foolish bugs related to unsigned ints, but now it is working.  I have tested **f**, **m**, and **l** with or without ,format and all works.  Good to go for approval, guys.

**Comment by PaulWessel** [comment]
Pinging @GenericMappingTools/core

File: doc/rst/source/movie.rst [hunk 1]
```diff
@@ -218,7 +218,8 @@ Optional Arguments
 **-M**\ [*frame*],[*format*]
```
Comment (meghanrjones) [inline doc/rst/source/movie.rst:218]:
```suggestion
**-M**\ [*frame*\|\ **f**\|\ **m**\|\ **l**],[*format*]
```

Comment (meghanrjones) [reply doc/rst/source/movie.rst:218]: Including the codes here and in the synopsis (L28) might make the new option more widely known.

**Review by meghanrjones — APPROVED** [review]
Looks good. I just made one documentation recommendation.

**Pull request merged**